testing/cloud/variables.tf

```
@@ -42,6 +42,12 @@ variable "elasticsearch_zone_count" {
   description = "Optional Elasticsearch zone count"
 }
 
+variable "observability_deployment" {
+  default     = "self"
+  type        = string
+  description = "Optional deployment ID, for platform observability. Use 'self' (the default) to send data to the same deployment, 'dedicated' to create a dedicated observability deployment, or 'none' to disable observability. Any other value is assumed to be the ID of an existing, externally managed Elasticsearch deployment."
```

Can we add a validation rule for this variable? Since the values are fixed (or deployment ID which is of length 32), and are expected to be one of a list, it fits the `validation` use case quite nicely: https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules.

```suggestion
  description = "Optional deployment ID, for platform observability. Use 'self' (the default) to send data to the same deployment, 'dedicated' to create a dedicated observability deployment, or 'none' to disable observability. Any other value is assumed to be the ID of an existing, externally managed Elasticsearch deployment."
  validation {
    condition     = can(regex("^self$|^dedicated$|^none$", var.observability_deployment)) || length(var.observability_deployment) == 32
    error_message = "invalid observability_deployment value. Must be one of 'self','dedicated','none', '<deployment id>'"
  }
```